src/create_disk.sh

```diff
@@ -30,14 +31,83 @@ Options:
     --help: show this help
     --kargs: kernel CLI args
     --no-x86-bios-bootloader: don't install BIOS bootloader on x86_64
+    --with-secure-execution:  generate qcow2 image with IBM SecureExecution feature enabled
 
 You probably don't want to run this script by hand. This script is
 run as part of 'coreos-assembler build'.
 EOC
 }
 
+#SecureExecution variables
+secure_execution_hostkey="/dev/disk/by-id/virtio-hostkey"
+secure_execution_initrd="/tmp/initrd.secure"
+secure_execution_bootfs_key="boot"
+secure_execution_bootfs_label="crypt_bootfs"
+secure_execution_bootfs_device="/dev/mapper/crypt_bootfs"
+
+secure_execution_rootfs_key="root"
+secure_execution_rootfs_label="crypt_rootfs"
+secure_execution_rootfs_device="/dev/mapper/crypt_rootfs"
+
+regenerate_initramfs() {
```

Comment:
Instead of unpacking and repacking the canonical initramfs, I think it'd be cleaner/safer to just copy it and append your new mini initramfs to it.

Reply:
doesn't work ((

Reply:
See e.g. https://github.com/coreos/coreos-assembler/blob/46e7d0c2efcf8f5c181115b8181c3dd9cd9a2f00/src/cmd-buildinitramfs-fast#L57-L59 for an example where we append to the canonical initramfs. Does `cosa buildinitramfs-fast` work for you on s390x?

Reply:
fixed. looks like `-mindepth 1` was the reason